ioc_copier/ioc_copier.py

```diff
@@ -0,0 +1,240 @@
+"""
+Script to handle duplicating IOCs
+"""
+
+from shutil import copytree
+import sys
+import os
+import re
+
+
+def rename_files(root_folder, rename, ioc):
+    """
+    Function to handle renaming of files.
+    Parameters:
+        root_folder - The root folder path for use in os.path.join.
+        rename - the path of the file or folder to rename.
+        padded_start_copy - the padded starting number, e.g. 01, 11 etc.
+        padded_current_copy - the padded version of the current ioc number.
+    """
+    if f"IOC_{padded_start_copy}" in rename:
+        os.rename(os.path.join(root_folder, rename),
+                  os.path.join(root_folder, rename.replace(f"IOC_{padded_start_copy}",
+                                                           f"IOC_{padded_current_copy}")))
+    if f"IOC-{padded_start_copy}" in rename:
+        os.rename(os.path.join(root_folder, rename),
+                  os.path.join(root_folder, rename.replace(f"IOC-{padded_start_copy}",
+                                                           f"IOC-{padded_current_copy}")))
+    if f"{ioc}_{padded_start_copy}" in rename:
+        os.rename(os.path.join(root_folder, rename),
+                  os.path.join(root_folder, rename.replace(f"{ioc}_{padded_start_copy}",
+                                                           f"{ioc}_{padded_current_copy}")))
+
+
+def replace_text(text_lines, ioc):
+    """
+    Function to handle replacing of text within files.
+    Parameters:
+        text_lines - the text from the file to process.
+        start - the value to look for to remove.
+        current - the value to replace start with.
+    """
+    for i in range(len(text_lines)):
```

Comment:
Could this be swapped out to use an enumerate instead of iterating with range and len?

Reply:
Pretty sure I tested that and it doesn't work, if I don't use the text_lines[i] style later it doesn't overwrite it within text_lines properly.

Reply:
so something like `for i, line in enumerate(text_lines):` wouldn't work?

Reply:
I mean, you can use that as the loop condition, but the `line` variable is useless, it doesn't work as an argument to the regex substitution, and it doesn't appear to be a reference to the line, so setting it to the new value doesn't overwrite the version of it actually in the list like doing `text_lines[i] = temp_text` does

Reply:
so I can change it to enumerate if you want, but baiscally its going to be `for i, _ in enumerate(text_lines):`

Reply:
hmmm, to handle this in a more pythonic way maybe I should pull out the loop internals into there own function and use a list comprehension instead?